### PR TITLE
test(ui): Convert perf landing to RTL

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This converts the performance landing page tests to RTL, and fixes a couple issues uncovered while converting the tests. Also adds a few key test ids to common points (grideditable), and fixes a bug with the initializeData setup with RTL.

#### Other:
- Removed "renders settings button for MEPS" test instead of converting it.

### CODE REVIEW

File: static/app/views/performance/table.tsx [hunk 1]
```diff
@@ -422,7 +422,7 @@ class _Table extends Component<Props, State> {
     const prependColumnWidths = ['max-content'];
 
     return (
-      <div>
+      <div data-test-id="PerformanceTable">
```
Comment (gggritso) [inline static/app/views/performance/table.tsx:425]: Why CamelCase here, but kabob-case in the other test IDs?

**Review by gggritso — COMMENTED** [review]
Sweet! One question: how does Sentry feel about using test IDs vs. roles? If the grid uses a real `<table>` element, you could probably use `getByRole("tablecell")` and so on

File: static/app/views/performance/table.tsx (hunk 1 above)
Comment (k-fish) [reply static/app/views/performance/table.tsx]: lol, will fix it.

**Comment by k-fish** [comment]
@gggritso I intentionally did not use `getByRole` despite it being better in various other ways, it's performance for large dom nodes was poor when I last tested it. I think once we add `getByRole` etc. to the testing transaction and checking in performance monitoring we can make the call either way if the performance impact is worth it. 

More details: https://github.com/getsentry/sentry/pull/32839#discussion_r832554205

**Review by gggritso — APPROVED** [review]
Legit 👌🏻

**Pull request merged**